swaybar/bar.c

```diff
@@ -256,33 +299,45 @@ void bar_setup(struct swaybar *bar,
 		bar->status = status_line_init(bar->config->status_command);
 	}
 
-	assert(bar->display = wl_display_connect(NULL));
+	bar->display = wl_display_connect(NULL);
+	assert(bar->display);
 
 	struct wl_registry *registry = wl_display_get_registry(bar->display);
 	wl_registry_add_listener(registry, &registry_listener, bar);
 	wl_display_roundtrip(bar->display);
 	assert(bar->compositor && bar->layer_shell && bar->shm);
+	wl_display_roundtrip(bar->display);
+
 	struct swaybar_pointer *pointer = &bar->pointer;
 
-	assert(pointer->cursor_theme = wl_cursor_theme_load(NULL, 16, bar->shm));
+	int max_scale = 1;
+	struct swaybar_output *output;
+	wl_list_for_each(output, &bar->outputs, link) {
+		if (output->scale > max_scale) {
+			max_scale = output->scale;
+		}
+	}
+
+	pointer->cursor_theme = wl_cursor_theme_load(
+				NULL, 16 * (max_scale * 2), bar->shm);
```

Comment:
Cursor size changes when hovering the bar. Should be `24 * max_scale`.

Reply:
What if the output scale is dynamically changed?

Reply:
Aha

Reply:
If the output scale is dynamically changed we're SOL, I want to file a ticket and deal with it never